codec-http2/src/main/java/io/netty/handler/codec/http2/DefaultHttp2Connection.java

```diff
@@ -131,18 +131,24 @@ final boolean isClosed() {
             promise.trySuccess(null);
             return promise;
         }
+
         Iterator<PrimitiveEntry<Http2Stream>> itr = streamMap.entries().iterator();
         // We must take care while iterating the streamMap as to not modify while iterating in case there are other code
         // paths iterating over the active streams.
         if (activeStreams.allowModifications()) {
```

Comment:
If we were to avoid **concurrent** modification, why we don't use AtomicInteger here to track concurrent modification?
Or perhaps we were just to avoid modification in recursive calls?